### PR TITLE
emiluaPlugins: introduce botan and secp256k1

### PR DESCRIPTION
**Pull request by manipuladordedados**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 342594` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>emiluaPlugins.botan</li>
    <li>emiluaPlugins.secp256k1</li>
  </ul>
</details>

**Review by lucasew — APPROVED** [review]

**Comment by lucasew** [comment]
@ofborg eval

**Review by thiagokokada — CHANGES_REQUESTED** [review]
Please squash your commits accordingly to [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md). We need to have 2 commits only:

```
emiluaPlugins.botan: init at 1.1.0
emiluaPlugins.secp256k1: init at 0.5.0
```

File: pkgs/development/emilua-plugins/botan/default.nix [hunk 1]
```diff
@@ -36,6 +36,7 @@ stdenv.mkDerivation rec {
     botan3
     luajit_openresty
     openssl
+    boost
```
Comment (lucasew) [inline pkgs/development/emilua-plugins/botan/default.nix:39]: This should be in the other commit

File: pkgs/top-level/emilua-plugins.nix [hunk 2]
```diff
@@ -11,5 +11,10 @@ emilua:
   beast = self.callPackage ../development/emilua-plugins/beast { };
   qt5 = self.callPackage ../development/emilua-plugins/qt5 { };
   qt6 = self.callPackage ../development/emilua-plugins/qt6 { };
-  botan = self.callPackage ../development/emilua-plugins/botan { };
+  botan = self.callPackage ../development/emilua-plugins/botan {
```
Comment (lucasew) [inline pkgs/top-level/emilua-plugins.nix:14]: Also in the other commit

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 342594` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>emiluaPlugins.botan</li>
    <li>emiluaPlugins.secp256k1</li>
  </ul>
</details>

**Review by lucasew — APPROVED** [review]

**Review by thiagokokada — APPROVED** [review]

File: pkgs/top-level/emilua-plugins.nix [hunk 3]
```diff
@@ -11,4 +11,10 @@ emilua:
   beast = self.callPackage ../development/emilua-plugins/beast { };
   qt5 = self.callPackage ../development/emilua-plugins/qt5 { };
   qt6 = self.callPackage ../development/emilua-plugins/qt6 { };
+  botan = self.callPackage ../development/emilua-plugins/botan {
```
Comment (SuperSandro2000) [inline pkgs/top-level/emilua-plugins.nix:14]: Shouldn't this list be sorted?

File: pkgs/development/emilua-plugins/secp256k1/default.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    fmt
+    secp256k1
+    luajit_openresty
+    openssl
+    boost
+  ];
+
+  nativeBuildInputs = [
+    gperf
+    gawk
```
Comment (SuperSandro2000) [inline pkgs/development/emilua-plugins/secp256k1/default.nix:44]: Isn't gawk part of stdenv?

File: pkgs/development/emilua-plugins/botan/default.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    fmt
+    botan3
+    luajit_openresty
+    openssl
+    boost
+  ];
+
+  nativeBuildInputs = [
+    gperf
+    gawk
```
Comment (SuperSandro2000) [inline pkgs/development/emilua-plugins/botan/default.nix:44]: Same

File: pkgs/development/emilua-plugins/secp256k1/default.nix (hunk 4 above)
Comment (manipuladordedados) [reply pkgs/development/emilua-plugins/secp256k1/default.nix:44]: Is?

Comment (lucasew) [reply pkgs/development/emilua-plugins/secp256k1/default.nix:44]: Try building it without gawk and see what happens

Comment (manipuladordedados) [reply pkgs/development/emilua-plugins/secp256k1/default.nix:44]: It is necessary, I tested it.

File: pkgs/top-level/emilua-plugins.nix (hunk 3 above)
Comment (manipuladordedados) [reply pkgs/top-level/emilua-plugins.nix:12]: Fixed.

**Review by lucasew — APPROVED** [review]

**Pull request merged**